OpenEdXMobile/src/main/java/org/edx/mobile/view/WebViewDiscoverCoursesFragment.java

```diff
@@ -37,10 +61,110 @@ public View onCreateView(LayoutInflater inflater, @Nullable ViewGroup container,
     public void onViewCreated(View view, Bundle savedInstanceState) {
         super.onViewCreated(view, savedInstanceState);
 
+        errorNotification = new FullScreenErrorNotification(binding.llContent);
+
         loadUrl(getInitialUrl());
+        if (environment.getConfig().getCourseDiscoveryConfig().isSubectDiscoveryEnabled()) {
+            initSubjects();
+        }
         EventBus.getDefault().register(this);
     }
 
+    private void initSubjects() {
+        final String subjectItemsJson;
+        try {
+            subjectItemsJson = FileUtil.loadTextFileFromResources(getContext(), R.raw.subjects);
+            final Type type = new TypeToken<List<SubjectModel>>() {
+            }.getType();
+            final List<SubjectModel> subjectModels = new Gson().fromJson(subjectItemsJson, type);
+            final List<SubjectModel> popularSubjects = new ArrayList<>();
+            for (SubjectModel subject : subjectModels) {
+                if (subject.type == SubjectModel.Type.POPULAR) {
+                    popularSubjects.add(subject);
+                }
+            }
+
+            final RecyclerView.Adapter adapter = new RecyclerView.Adapter() {
```

Comment:
This is done on purpose, I am planning to try this library for RecyclerViewAdapter and see if it works as intended and reduces boiler plate code:
https://github.com/gurleensethi/LiteUtilities